### PR TITLE
PEOPLE: Bemerkungsfelder in API

### LINKED ISSUE #518: PEOPLE: Bemerkungsfelder in API
~blocked by #511~

Mit #511 werden auf `Person` 6 neue Bemerkungsfelder hinzugefügt: **"Bemerkung Sektion 1” … "Bemerkung Sektion 5”** sowie **"Bemerkung Geschäftsstelle"**
Diese Felder sollen im JSON:API ebenfalls zurückgegeben werden.

* Die Werte der Felder **"Bemerkung Sektion 1” … "Bemerkung Sektion 5”** dürfen nur sichtbar sein, wenn der user die Berechtigung `:manage_national_office_remark` auf der Person hat.
* Das Werte der Felder **"Bemerkung Geschäftsstelle"** dürfen nur sichtbar sein, wenn der user die Berechtigung `:manage_section_remarks` auf der Person hat.

# Tech-Spec

* in `SacCas::PersonResource` neue Attribute definieren:
  ```ruby
    with_options(writable: false) do
      attribute :remark_head_office, readable: :can_read_remark_head_office?

      (1..5).each do |num|
        attribute "remark_sektion_#{num}", readable: :can_read_remarks_sektion?
      end
    end
  ```

* guard Methoden:

  ```ruby
    def can_read_remark_head_office?(person) = can?(:manage_remark_head_office, person)
    def can_read_remarks_sektion?(person) = can?(:manage_remarks_sektion, person)
  ```
  
* schema.json aktualisieren: `rails graphiti:schema:generate` und resultierendes `schema.json` einchecken

# ToDo

* [x] Attribute in PersonResource einbauen
* [x] schema.json aktualisieren und einchecken
* [x] Specs schreiben

### PR DESCRIPTION
**Pull request by hunchr**
Fixes #518

- Testbar auf z.B. http://localhost:3000/api/people/600780
- Wenn man keine Berechtigungen hat, ein `remark` feld zu sehen, wird dies in der API trotzdem angezeigt, aber immer mit `null` als Wert

### CODE REVIEW

**Comment by rnestler** [comment]
Ich würde request specs für APIs erwarteten, welche die Daten überprüfen die zurückgegeben werden.

File: spec/api/people/show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2024, Schweizer Alpen-Club. This file is part of
+#  hitobito and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_sac_cas
+
+require "spec_helper"
+
+RSpec.describe "people#show", type: :request do
```
Comment (rnestler) [inline spec/api/people/show_spec.rb:10]: If you move to file to `spec/requests/api/...` then you also don't need to specify the `type:`. Also I'd use the conroller class in `describe`

```suggestion
RSpec.describe FoobarController do
```
On the other hand the existing api specs are done exactly like this... So lets keep it.

File: spec/api/people/show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  let(:token) { service_tokens(:permitted_root_layer_token).token }
+  let(:person) { people(:admin) }
+
+  subject(:make_request) { jsonapi_get "/api/people/#{person.id}" }
+
+  before do
+    person.update!(sac_remark_national_office: "Remark", sac_remark_section_1: "Remark")
+    sign_in(person)
```
Comment (rnestler) [inline spec/api/people/show_spec.rb:18]: Shouldn't you use the token instead of signing in? I think you should be using `it_behaves_like "jsonapi authorized requests" do` like in the other  api specs.

File: spec/api/people/show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+#  Copyright (c) 2024, Schweizer Alpen-Club. This file is part of
+#  hitobito and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_sac_cas
+
+require "spec_helper"
+
+RSpec.describe "people#show", type: :request do
+  let(:token) { service_tokens(:permitted_root_layer_token).token }
```
Comment (rnestler) [inline spec/api/people/show_spec.rb:11]: You aren't using the token anywere.

File: spec/api/people/show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  let(:person) { people(:admin) }
+
+  subject(:make_request) { jsonapi_get "/api/people/#{person.id}" }
```
Comment (rnestler) [inline spec/api/people/show_spec.rb:14]: I didn't know about `jsonapi_get`. Nice :+1:

**Comment by hunchr** [comment]
todo: use the token instead of signing in

File: spec/api/people/show_spec.rb (hunk 2 above)
Comment (hunchr) [reply spec/api/people/show_spec.rb]: i don't know how the token is supposed to work, but that'd be better

File: spec/support/graphiti/schema.json [hunk 5]
```diff
@@ -2450,17 +2486,6 @@
...
-        "membership_number": {
-          "type": "integer",
-          "operators": [
-            "eq",
-            "not_eq",
-            "gt",
-            "gte",
-            "lt",
-            "lte"
-          ]
```
Comment (rnestler) [inline spec/support/graphiti/schema.json:2463]: Why was this removed?

Comment (hunchr) [reply spec/support/graphiti/schema.json:2463]: there was a typo in [filterable](https://github.com/hitobito/hitobito_sac_cas/pull/711/files#diff-89b0aa447b4c15f373fe22729e9b5456a4333339615c366af5af1e3b760c20e4L13)

**Review by rnestler — APPROVED** [review]
